src/AppInstallerRepositoryCore/Microsoft/Schema/1_4/DependenciesTable.cpp

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "DependenciesTable.h"
+#include "SQLiteStatementBuilder.h"
+#include "winget\DependenciesGraph.h"
+#include "Microsoft/Schema/1_0/OneToOneTable.h"
+#include "Microsoft/Schema/1_0/IdTable.h"
+#include "Microsoft/Schema/1_0/ManifestTable.h"
+#include "Microsoft/Schema/1_0/VersionTable.h"
+#include "Microsoft/Schema/1_0/Interface.h"
+#include "Microsoft/Schema/1_0/ChannelTable.h"
+
+namespace AppInstaller::Repository::Microsoft::Schema::V1_4
+{
+	using namespace AppInstaller;
+	using namespace std::string_view_literals;
+	using namespace SQLite::Builder;
+	using namespace Schema::V1_0;
+	using QCol = SQLite::Builder::QualifiedColumn;
+
+	static constexpr std::string_view s_DependenciesTable_Table_Name = "dependencies"sv;
+	static constexpr std::string_view s_DependenciesTable_Index_Name = "dependencies_index"sv;
+	static constexpr std::string_view s_DependenciesTable_Manifest_Column_Name = "manifest"sv;
+	static constexpr std::string_view s_DependenciesTable_MinVersion_Column_Name = "min_version"sv;
+	static constexpr std::string_view s_DependenciesTable_PackageId_Column_Name = "package_id";
+
+	namespace 
+	{
+		std::vector<Manifest::Dependency> GetDependencies(
+			const Manifest::Manifest& manifest, Manifest::DependencyType dependencyType)
+		{
+			std::vector<Manifest::Dependency>  dependencies;
+
+			for (const auto& installer : manifest.Installers)
+			{
+				installer.Dependencies.ApplyToAll([&](Manifest::Dependency dependency)
+					{
+						if (dependency.Type == dependencyType)
+						{
+							dependencies.emplace_back(dependency);
+						}
+					});
+			}
+
+			return dependencies;
+		}
+
+		void RemoveDependenciesByRowIds(SQLite::Connection& connection, std::vector<SQLite::rowid_t> dependencyRowId)
+		{
+			using namespace SQLite::Builder;
+			SQLite::Savepoint savepoint = SQLite::Savepoint::Create(connection, std::string{ s_DependenciesTable_Table_Name } + "remove_dependencies_by_rowid");
+
+			if (dependencyRowId.size() > 0)
```

Comment:
If you are going to check the size, just check for `empty` and get out up front rather than creating a savepoint.